### PR TITLE
Add Papua New Guinea holidays

### LINKED ISSUE #1239: Add Papua New Guinea holidays
Links:
  - https://en.wikipedia.org/wiki/Public_holidays_in_Papua_New_Guinea
  - https://www.timeanddate.com/holidays/papua-new-guinea/

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

Add Papua New Guinea holidays.

Closes #1239 .

## Type of change

- [x] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/countries/papua_new_guinea.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+        # Section 3: Anniversary of Birthday of the Queen/King.
+        # In Papua New Guinea, it is usually celebrated on the second Monday of June every year.
+        # From 2023, the date changed to Jun 17 (Special Observed for 2023 on Jun 16).
+        # For localization, see "self.sovereign_birthday".
+
+        if self._year <= 2022:
+            self._add_holiday_2nd_mon_of_jun(self.sovereign_birthday)
+        else:
+            self._add_holiday_jun_17(self.sovereign_birthday)
```
Comment (KJhellico) [inline holidays/countries/papua_new_guinea.py:114]:
```suggestion
        if self._year <= 2022:
            # Queen's Birthday.
            self._add_holiday_2nd_mon_of_jun("Queen's Birthday")
        else:
            # King's Birthday.
            self._add_holiday_jun_17("King's Birthday")
```
I think special property is a bit excessive here.

**Review by KJhellico — COMMENTED** [review]
New countries it's always great! 👍

File: holidays/countries/papua_new_guinea.py (hunk 1 above)
Comment (PPsyrius) [reply holidays/countries/papua_new_guinea.py:114]: Fixed.

File: holidays/countries/papua_new_guinea.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+      - 2023 Sir Rabbie Namaliu's Funeral: https://pnghausbung.com/breaking-public-holiday-tomorrow-in-respect-of-late-sir-rabbies-state-funeral/  # noqa: E501
+
+    * Nov 15, 2018 is an APEC Holiday too, but for Port Moresby-only: https://www.businessadvantagepng.com/public-holidays-declared-in-port-moresby-due-to-apec-summit-says-ceo/  # noqa: E501
+    """
+
+    # National Day of Mourning for Sir Michael Somare.
+    michael_somare_mourning = "National Day of Mourning for Sir Michael Somare"
+
+    # Grand Chief Sir Michael Somare Remembrance Day
+    somare_remembrance_day = "Grand Chief Sir Michael Somare Remembrance Day"
```
Comment (arkid15r) [inline holidays/countries/papua_new_guinea.py:151]: Could use some consistency here:
`michael_somare_mourning` -> `sir_michael_somare_mourning_day`
`somare_remembrance_day` -> `sir_michael_somare_remembrance_day`

**Review by arkid15r — COMMENTED** [review]
The 🇵🇬 code looks good, just a few minor comments here:

File: holidays/countries/papua_new_guinea.py (hunk 2 above)
Comment (PPsyrius) [reply holidays/countries/papua_new_guinea.py]: Fixed.

**Review by KJhellico — COMMENTED** [review]
LGTM.

**Review by arkid15r — APPROVED** [review]

**Pull request merged**